server/app/assets/javascripts/accordion.ts

```diff
@@ -1,37 +1,74 @@
 /** This class is a simple controller for accordion elements. */
-class AccordionController {
-  static accordionSelector = '.cf-accordion'
-  static accordionButtonSelector = '.cf-accordion-button'
-  static accordionHeaderSelector = '.cf-accordion-header'
-  static accordionVisibleClass = 'cf-accordion-visible'
-
-  constructor() {
-    this.init()
-  }
-
-  private init() {
-    const items = Array.from(
-      document.querySelectorAll(AccordionController.accordionHeaderSelector),
-    )
-
-    items.forEach((item) => {
-      item.addEventListener('click', AccordionController.toggleAccordion)
-    })
-  }
-
-  static toggleAccordion(event: Event) {
-    const target = event.target as Element
-    const parentAccordion = target.closest(
-      AccordionController.accordionSelector,
-    )
-    if (parentAccordion) {
-      parentAccordion.classList.toggle(
-        AccordionController.accordionVisibleClass,
-      )
+export class AccordionController {
+  buttonNode: HTMLElement
+  controlledNode: HTMLElement | null
+
+  /*
+   *   This content is licensed according to the W3C Software License at
+   *   https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
+   *
+   *   Desc:   Disclosure button widget that implements ARIA Authoring Best Practices
+   */
+  constructor(buttonNode: HTMLElement) {
+    this.buttonNode = buttonNode
+    this.controlledNode = null
+    const id = this.buttonNode.getAttribute('aria-controls')
+
+    if (id) {
+      this.controlledNode = document.getElementById(id)
+    }
+
+    this.buttonNode.setAttribute('aria-expanded', 'false')
+    this.hideContent()
+
+    this.buttonNode.addEventListener('click', this.onClick.bind(this))
+    this.buttonNode.addEventListener('focus', this.onFocus.bind(this))
+    this.buttonNode.addEventListener('blur', this.onBlur.bind(this))
+  }
+
+  showContent() {
+    if (this.controlledNode) {
+      this.controlledNode.style.display = 'block'
+    }
+  }
+
+  hideContent() {
+    if (this.controlledNode) {
+      this.controlledNode.style.display = 'none'
+    }
+  }
+
+  toggleExpand() {
+    if (this.buttonNode.getAttribute('aria-expanded') === 'true') {
+      this.buttonNode.setAttribute('aria-expanded', 'false')
+      this.hideContent()
+    } else {
+      this.buttonNode.setAttribute('aria-expanded', 'true')
+      this.showContent()
     }
   }
+
+  /* EVENT HANDLERS */
+
+  onClick() {
+    this.toggleExpand()
+  }
+
+  onFocus() {
+    this.buttonNode.classList.add('focus')
+  }
+
+  onBlur() {
+    this.buttonNode.classList.remove('focus')
+  }
 }
 
 export function init() {
-  new AccordionController()
+  const disclosureButtons: NodeListOf<Element> = document.querySelectorAll(
```

Comment:
question(non-blocking): Just curious why you restructured it this way instead of keeping the exported `init` function only responsible for creating the instance. There seems to be a pattern in the other files of keeping the exported `init` function very minimal, but I don't have a strong preference about this.

Reply:
The idea was to only initialize this class when accordion functionality is needed, which can slightly improve page load performance.

Reply:
Nice! Good idea.